python/orca/src/bigdl/orca/learn/tf2/pyspark_estimator.py

```diff
@@ -108,8 +108,18 @@ def __init__(self,
             self.log_server_thread = start_log_server(self.ip, self.port)
 
     def _get_cluster_info(self, sc):
-        cluster_info = self.workerRDD.barrier().mapPartitions(find_ip_and_free_port).collect()
-        return cluster_info
+        def get_worker_address(iter):
+            worker_ip = get_node_ip()
```

Comment:
Remark: using this function can get the k8s virtual ip while using Sparkbarriercontext can only get the physical node ip